mlflow/data_mlflow_experiment.go

```diff
@@ -0,0 +1,42 @@
+package mlflow
+
+import (
+	"context"
+	"fmt"
+	"github.com/databricks/databricks-sdk-go/service/ml"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceExperiment() *schema.Resource {
+	return common.WorkspaceData(func(ctx context.Context, data *struct {
```

Comment:
Can you just use *ml.Experiment from the Go SDK?

Reply:
At this moment it is not possible to annotate Terraform fields with the Go SDK model so this approach is not possible. Current approach is fine for now.

I did change `id` to `experiment_id` to make the transition to the Go SDK model backwards compatible.

Relevant issue for annotation support: #2224